libs/community/langchain_community/chat_models/mistral.py

```diff
@@ -0,0 +1,31 @@
+from typing import List
+
+from langchain_core.messages import (
+    AIMessage,
+    BaseMessage,
+    ChatMessage,
+    HumanMessage,
+    SystemMessage,
+)
+
+
+def _convert_one_message_to_text_mistral(message: BaseMessage) -> str:
+    if isinstance(message, ChatMessage):
+        message_text = f"\n\n{message.role.capitalize()}: {message.content}"
+    elif isinstance(message, HumanMessage):
+        message_text = f"[INST] {message.content} [/INST]"
+    elif isinstance(message, AIMessage):
+        message_text = f"{message.content}"
+    elif isinstance(message, SystemMessage):
+        message_text = f"<<SYS>> {message.content} <</SYS>>"
+    else:
+        raise ValueError(f"Got unknown type {message}")
+    return message_text
+
+
+def convert_messages_to_prompt_mistral(messages: List[BaseMessage]) -> str:
+    """Convert a list of messages to a prompt for mistral."""
+
+    return "\n".join(
+        [_convert_one_message_to_text_mistral(message) for message in messages]
+    )
```

Comment:
I do not recommend adding anything in the `mistral.py` files. Rather, just copy these to the bedrock code.
```suggestion

```

Reply:
thanks @3coins !
I deleted `mistral.py` and moved the code to `bedrock.py `
I've also added the code for async streaming support

I've tested and everything seems to work :)